### PR TITLE
[Bugfix] Apply additive penalties during DFlash verify

### LINKED ISSUE #33493: [Bug] DSPARK retrieves wrong key "acc_linear_penalities" from sampling_info
### Checklist

- [x] I searched related issues but found no solution.
- [x] The bug persists in the latest version.
- [x] Issues without environment info and a minimal reproducible demo are hard to resolve and may receive no feedback.
- [x] If this is not a bug report but a general question, please start a discussion at https://github.com/sgl-project/sglang/discussions. Otherwise, it will be closed.
- [x] Please use English. Otherwise, it will be closed.

### Describe the bug

https://github.com/sgl-project/sglang/blob/main/python/sglang/srt/speculative/dflash_utils.py#L147
```
acc_linear_penalties = getattr(sampling_info, "acc_linear_penalties", None)
```

While there's no `acc_linear_penalties` in [sampling_info](https://github.com/sgl-project/sglang/blob/main/python/sglang/srt/sampling/sampling_batch_info.py#L29)

Should change to 

```
--- a/python/sglang/srt/speculative/dflash_utils.py
+++ b/python/sglang/srt/speculative/dflash_utils.py
@@ -143,7 +143,7 @@
             num_tokens_in_batch=draft_token_num,
         )
 
-    acc_linear_penalties = getattr(sampling_info, "acc_linear_penalties", None)
+    acc_linear_penalties = getattr(sampling_info, "acc_additive_penalties", None)
     penalizer = getattr(sampling_info, "penalizer_orchestrator", None)
     vocab_mask = getattr(sampling_info, "vocab_mask", None)
     logit_bias = getattr(sampling_info, "logit_bias", None)
--- a/python/sglang/srt/speculative/dspark_components/dspark_verify.py
+++ b/python/sglang/srt/speculative/dspark_components/dspark_verify.py
@@ -44,7 +44,7 @@
         return True
     if sampling_info.has_custom_logit_processor:
         return False
-    if getattr(sampling_info, "acc_linear_penalties", None) is not None:
+    if getattr(sampling_info, "acc_additive_penalties", None) is not None:
         return False
     penalizer = getattr(sampling_info, "penalizer_orchestrator", None)
     if penalizer is not None and penalizer.is_required:
```

### Reproduction

Start sglang with dspark:

```
sglang serve \
  --trust-remote-code \
  --model-path deepseek-ai/DeepSeek-V4-Flash-0731 \
  --tp 4 \
  --moe-runner-backend flashinfer_mxfp4 \
  --speculative-algorithm DSPARK \
  --mem-fraction-static 0.88 \
  --host 0.0.0.0 \
  --port 30000
```

Send request setting `min_new_tokens`, but not work
```
{
  "model": "deepseek-ai/DeepSeek-V4-Flash-0731",
  "messages": [{"role": "user", "content": "Reply with exactly one short sentence."}],
  "max_tokens": 80,
  "min_tokens": 50,
  "temperature": 0
}
```

Live repro result:
HTTP 200
finish_reason: stop
completion_tokens: 18
content: "The sun dipped below the horizon, painting the sky in shades of amber and rose."

Expected completion_tokens >= 50 with min_tokens=50, got: 3

### Environment

sglang v0.5.16, also in main branch

### PR DESCRIPTION
**Pull request by chilltongx**
## Summary

- read the existing `SamplingBatchInfo.acc_additive_penalties` field during DFlash verify
- keep the DSPARK no-adjustment fast path disabled when accumulated additive penalties are present
- add focused CPU regression coverage for verify-block broadcasting and fast-path selection

## Root cause

`SamplingBatchInfo` stores overlap-mode additive penalties in `acc_additive_penalties`, but the DFlash verify path and DSPARK fast-path predicate looked up the nonexistent `acc_linear_penalties` attribute. As a result, accumulated penalties such as the minimum-new-token EOS suppression could be skipped during DSPARK verification.

## Impact

Requests using overlap-mode additive penalties now apply the per-request adjustment to every token in the DFlash/DSPARK verify block. Requests without adjustments retain the existing fast path.

## Validation

- focused CPU pytest: `3 passed`
- Black, isort, Ruff (`F401,F821,UP037`), `py_compile`, and `git diff --check`: passed

Fixes #33493

---
### CI States

Latest PR Test (Base): :x: [Run #34680548624](https://github.com/sgl-project/sglang/actions/runs/34680548624)
Latest PR Test (Extra): :x: [Run #34680548471](https://github.com/sgl-project/sglang/actions/runs/34680548471)
Latest PR Test (AMD ROCm 10): :x: [Run #34680548509](https://github.com/sgl-project/sglang/actions/runs/34680548509)

### CODE REVIEW

**Comment by chilltongx** [comment]
This PR is now ready for review. The existing required CI failure is from the initial draft-state gate, not a test failure. Could a maintainer please add the run-ci label, rerun CI, and review when convenient? Thank you!

**Comment by chilltongx** [comment]
Rebased onto the latest `main` (4cb5aebf). The focused CPU suite passes: `test_dflash_verify_logits_adjustments.py` — 3 passed; affected-file pre-commit also passes. Ready for the `run-ci` label and review.

**Comment by chilltongx** [comment]
Rebased onto current `main` (`cbd02715`) and force-pushed an unchanged patch; `git range-diff b92df6e8^! 30b4d1bd^!` reports the commit as equivalent.\n\nFresh validation:\n- focused CPU suite: **3 passed**\n- affected-file pre-commit: passed\n- `git diff --check`: passed\n- speculative naming audit: touched identifiers use the current canonical `acc_additive_penalties`; no new `accepted_*`/token-id naming violations\n\nCould a maintainer please review and add `run-ci` when convenient?

**Comment by chilltongx** [comment]
Updated to current `main` (`6ba96d32`), including the required `MIN_BASE_SHA` (`3700c4ee`). Current head: `2aa83bba`.

The focused CPU regressions pass: **3 tests**, covering additive-penalty broadcasting over the verify block and the corresponding no-op fast path. `git range-diff` confirms that the original patch is unchanged.

All applicable local pre-commit hooks pass. Could a maintainer please review and add `run-ci` for upstream test coverage?
